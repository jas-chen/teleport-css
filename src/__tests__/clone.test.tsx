import { expect, test } from 'vitest';
import { renderToStaticMarkup } from 'react-dom/server';
import { MouseEvent } from 'react';
import fnv1a from '@sindresorhus/fnv1a';
import { create } from '..';
import pretty from 'pretty';

function hashFn(value: string) {
  return fnv1a(value, { size: 64 }).toString(36).slice(0, 8);
}

const { styled, cloneAs } = create({
  hashFn,
});

test('can clone style', () => {
  const Button = styled('button', () => ({
    color: 'red',
    background: 'blue',
  }));

  const Link = cloneAs(Button, 'a');

  expect(
    pretty(
      renderToStaticMarkup(
        <div>
          <Button
            onClick={
              // test typing
              (e: MouseEvent<HTMLButtonElement>) => {
                console.log(e);
              }
            }
          />
          <Link
            onClick={
              // test typing
              (e: MouseEvent<HTMLAnchorElement>) => {
                console.log(e);
              }
            }
          />
        </div>,
      ),
    ),
  ).toMatchSnapshot();
});
