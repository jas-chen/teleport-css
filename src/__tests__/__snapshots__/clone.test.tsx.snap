// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`can clone style 1`] = `
"<style data-precedence="1" data-href="x-157mpepm1 x-39u6vegt1">
  .x-157mpepm1 {
    color: red
  }

  .x-39u6vegt1 {
    background: blue
  }
</style>
<div><button class="x-157mpepm1 x-39u6vegt1"></button><a class="x-157mpepm1 x-39u6vegt1"></a></div>"
`;
